tests/jballerina-unit-test/src/test/resources/test-src/functions/cyclic-functions-negative.bal

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+final readonly & map<json|error> func = {
+    "f1": f1()
```

Comment:
I would add the test without the call too. To ensure the current behaviour is tested. So two tests, one which may no longer result in an error in the future.

Reply:
https://github.com/ballerina-platform/ballerina-lang/blob/3d985ba1a71292f872ed5a1a1e21bf672fce047a/tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/functions/test_different_function_signatures.bal#L336
Here we have a test like that.

Reply:
Not a negative test though?

Reply:
Added a test